src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -289,6 +289,11 @@ public async Task<object> GetDescriptionAsync(IAsyncCompletionSession session, V
             CompletionService completionService,
             Dictionary<string, AsyncCompletionData.CompletionFilter> filterCache)
         {
+            if (roslynItem.UseEditorCompletionItemCache && roslynItem.CachedEditorCompletionItem is VSCompletionItem vsItem)
```

Comment:
if we are using the filterCache pattern, why do we use different pattern for roslynItem? can it pass in vsitemCache as well?

roslyn item lives multiple completion session? is that the reason?

Reply:
I see. this actually outlives multiple completion session. if we are going to do this, shouldn't we do same for things like snippet, keyword, operators and etc?

Reply:
We can do the same for those other kind of completion items as well, but their number is usually small so it probably won't make any difference.

Reply:
Agree with gen.  The major value is with type imports which need to be fast, not churn memory, but represent many thousands of items.